MagicStorageConfig.cs

```diff
@@ -31,6 +34,10 @@ public class MagicStorageConfig : ModConfig
 		[DefaultValue(false)]
 		public bool recipeBlacklist;
 
+		[JsonProperty(ItemConverterType = typeof(ItemDefinitionToFromStringJsonConverter))]
```

Comment:
I'm not too familiar with configs so could you tell me what this does?

Reply:
This is only done because the default format for serializing `ItemDefinition` to JSON is to include all fields (this includes Name, DisplayName and Type). This makes the configuration files huge and redundant. A custom JSON format conversion is used here to store only a string, just like Ore Excavator mod.

Reply:
Would it make more sense to use our own class instead of hacking around `ItemDefinition`?

Reply:
Do you mean inheriting from `ItemDefinition` , or a wrapper class for `HashSet<ItemDefinition>` ? I think both solutions are less concise than using a custom converter. This is also a general solution for JSON custom formats.

Reply:
Are you sure `Type` and `DisplayName` are included in the serialized json?

Reply:
yes, although it behaves normally when an ItemDefinition is used as the Key of a dictionary, the behavior is inconsistent in a HashSet. You can try removing Attribute and observe the saved data structure.

Reply:
This is actually a tml bug, missing JsonIgnore attribute on those properties.
Please add a todo comment that says to remove this when tml add the attributes

Reply:
> This is actually a tml bug, missing `JsonIgnore` attribute on those properties. Please add a todo comment that says to remove this when tml adds the attributes

This may require a custom type converter to resolve. Directly ignoring unnecessary fields will still be partially redundant. But we do need to add a TODO.